geopandas/tools/tests/test_sjoin.py

```diff
@@ -119,21 +119,39 @@ def test_left(self, op, dfs):
 
     def test_empty_join(self):
         # Check empty joins
-        polygons = geopandas.GeoDataFrame({'col2': [1, 2], 
-                                           'geometry':  [Polygon([(0, 0), (1, 0), 
-                                                                  (1, 1), (0, 1)]), 
-                                                         Polygon([(1, 0), (2, 0), 
-                                                                  (2, 1), (1, 1)])
+        polygons = geopandas.GeoDataFrame({'col2': [1, 2],
+                                           'geometry': [Polygon([(0, 0), (1, 0),
+                                                                 (1, 1), (0, 1)]),
+                                                        Polygon([(1, 0), (2, 0),
+                                                                 (2, 1), (1, 1)])
                                                         ]})
-        not_in = geopandas.GeoDataFrame({'col1': [1], 
-                             'geometry': [Point(-0.5, 0.5)]})
+        not_in = geopandas.GeoDataFrame({'col1': [1],
+                                         'geometry': [Point(-0.5, 0.5)]})
         empty = sjoin(not_in, polygons, how='left', op='intersects')
         assert empty.index_right.isnull().all()
         empty = sjoin(not_in, polygons, how='right', op='intersects')
         assert empty.index_left.isnull().all()
         empty = sjoin(not_in, polygons, how='inner', op='intersects')
         assert empty.empty
 
+    @pytest.mark.parametrize('dfs', ['default-index', 'string-index'],
+                             indirect=True)
+    def test_sjoin_invalid_right(self, dfs):
+        index, df1, df2, expected = dfs
+        with pytest.raises(ValueError) as excinfo:
+            res = sjoin(df1, df2.geometry)
+
+        assert '`right_df` should be GeoDataFrame' in str(excinfo.value)
```

Comment:
You can also do this with the `match` keyword to `with pytest.raises(ValueError, match=".."): ...`